stem/model.py

```diff
@@ -1,24 +1,164 @@
-from typing import List
+from typing import List, Sequence, Dict, Any, Optional, Union
+
+from gmsh_utils import gmsh_IO
 
 from stem.model_part import ModelPart, BodyModelPart
+from stem.soil_material import *
+from stem.structural_material import *
+from stem.geometry import Geometry
 
 
 class Model:
     """
     A class to represent the main model.
 
     Attributes:
+        - ndim (int): Number of dimensions of the model
         - project_parameters (dict): A dictionary containing the project parameters.
         - solver (Solver): The solver used to solve the problem.
-        - body_model_parts (list): A list containing the body model parts.
-        - process_model_parts (list): A list containing the process model parts.
+        - geometry (Optional[:class:`stem.geometry.Geometry`]) The geometry of the whole model.
+        - body_model_parts (List[BodyModelPart]): A list containing the body model parts.
+        - process_model_parts (List[ModelPart]): A list containing the process model parts.
+        - extrusion_length(Optional[Sequence[float]]): The extrusion length in x,y and z direction
```

Comment:
- extrusion_length (Optional[Sequence[float]]): The extrusion length in x, y and z direction